code/a/analyze_nl_boot.do

```diff
@@ -0,0 +1,138 @@
+  append_est_to_file using $tmpfile, b(ln_light2012) s(sdxc_cons,`b') 
+}
+
+/********************/
+/* town-level stats */
+/********************/
+
+/* open shrid data */
+use $tmp/shrug_combined, clear
+group pc11_state_id
+drop if pc11_sector == 2
+foreach v in $tvars {
+  drop if mi(`v')
+}
+
+/* keep only the vars we use for the regs */
+keep shrid ln_light2012 ln_num_cells sdgroup ln_consumption
+
+/* replace consumption data with bootstrap data B */
+merge 1:1 shrid using $secc/parsed_draft/dta/ancillary/shrug_urban_cons_boot, keepusing(secc_cons_pc_*) keep(match) nogen
+
+/* loop over bootstraps */
+di "Bootstrapping towns..."
+qui forval b = 1/$nboot {
+
+  noi di `b'
+  
+  /* set consumption to the current bootstrap value */
+  replace ln_consumption = ln(secc_cons_pc_`b')
+  
+  /* run the regressions and store the bootstrap coef */
+  quireg ln_consumption ln_light2012 ln_num_cells, robust title(ln_consumption) absorb(sdgroup)
+  append_est_to_file using $tmpfile, b(ln_light2012) s(txc_d_cons,`b') 
+
+}
+
+/***********************/
+/* village-level stats */
+/***********************/
+
+/* open shrid data */
+use $tmp/shrug_combined, clear
+keep if pc11_sector == 2
+foreach v in $vvars {
+  drop if mi(`v')
+}
+
+/* keep only the vars we use for the regs */
+keep shrid ln_light2012 ln_num_cells sdgroup sdsgroup ln_consumption
+
+/* replace consumption data with bootstrap data B */
+merge 1:1 shrid using $secc/mord/dta/ancillary/shrug_rural_cons_boot, keepusing(secc_cons_pc_*) keep(match) nogen
+
+/* loop over bootstraps */
+di "Bootstrapping villages..."
+qui forval b = 1/$nboot {
+
+  noi di `b'
+  
+  /* set consumption to the current bootstrap value */
+  replace ln_consumption = ln(secc_cons_pc_`b')
+  
+  /* run the regressions and store the bootstrap coef */
+  quireg ln_consumption ln_light2012 ln_num_cells, robust title(ln_consumption) absorb(sdgroup)
+  append_est_to_file using $tmpfile, b(ln_light2012) s(vxc_d_cons,`b') 
+
+  quireg ln_consumption ln_light2012 ln_num_cells, robust title(ln_consumption) absorb(sdsgroup)
+  append_est_to_file using $tmpfile, b(ln_light2012) s(vxc_sd_cons,`b')
+}
+
+/* get the mean and standard deviation from the bootstrapped data */
+import delimited using $tmpfile, clear
+
+/* copy the template file from the non-boot regression */
+global output_csv $out/nl_xc_boot.csv
+shell cp $out/nl_xc.csv $output_csv
+
+/* store the stats for each measure */
+foreach stat in vxc_d vxc_sd txc_d {
+  sum b if var == "`stat'_cons"
+  local beta: di %5.3f `r(mean)'
+  local se: di %5.3f `r(sd)'
+  local t = `beta' / `se'
+  local p = 2 * (1 - normal(`t'))
+  count_stars, p(`p')
+  local stars `r(stars)'
+  local starbeta `beta'`stars'
+  
+  /* insert the stats into the output file */
+  insert_into_file using $output_csv, key(`stat'_cons_starbeta) val("`starbeta'")
+  insert_into_file using $output_csv, key(`stat'_cons_se) val(`se')
+}
+
+shell diff $out/nl_xc.csv $output_csv
+
```

Comment:
you can delete the `diff` - this was an internal check

Reply:
fixed in this [commit](https://github.com/devdatalab/paper-shrug/commit/ac6b4450ab5172fc6e95e16118c4a0b0ffefd180)